storage/redis_adapter.go

```diff
@@ -42,20 +44,20 @@ func (mp *messagePipeliner) addMessage(message *message) {
 		mp.queuedApps[message.app] = true
 		mp.messageCount++
 	} else {
-		log.Printf("Error adding rpush to %s to the pipeline: %s", message.app, err)
+		mp.errCh <- fmt.Errorf("Error adding rpush to %s to the pipeline: %s", message.app, err)
```

Comment:
it would be helpful to close `errCh` after you use it (see [below](https://github.com/deis/logger/pull/100/commits/8dbb7e66c0520185a9568d84564555552c5e3a22#r71035722))